### PR TITLE
improve initial installation

### PR DESCRIPTION
**Pull request by scoopex**
- add option for package pinning on debian and ubuntu to prevent
  accidential upgrades
- make dashboard installation more reliable
- simple reformatting to satisfy formatting rules
- adapt an enhance tests for the changes above
- add /usr/bin to default_path to solve #415

Sorry, i was not successful in executing the litmus tests described in the "Development" section on my ubuntu 18.04 system. "litmus:install_module" dropped very cryptic messages which were not understandable for a litmus newbie....
So probably you can fix potential acceptance test problems to get these useful changes to upstream?

### CODE REVIEW

File: manifests/packages.pp [hunk 1]
```diff
@@ -78,11 +80,26 @@
...
+          }else {
+            file { '/etc/apt/preferences.d/docker':
+              ensure => absent,
+            }
+          }
+        }else {
           package { $docker_package_name:
             ensure => $docker_version,
           }
+          if $pin_packages {
```
Comment (sanfrancrisko) [inline manifests/packages.pp:100]: Can you help me understand this condition - if someone has **not** set the `create_repos` and `manage_docker` parameters, indicating that they do not want to install and manage Docker, would we even need to fail if `pin_packages` is set? 

I guess it is good to warn and indicate that the configuration you set is not going to be taken in to consideration - would that suffice or is there something I'm missing?

Comment (scoopex) [reply manifests/packages.pp:100]: Your are right, safety was the intention of the fail() message.
Accidentally i missed to describe this in the fail() message - i improved that now.

**Comment by scoopex** [comment]
> I'll take a look at testing these changes on Ubuntu 18.04 on our internal infrastructure. I would be keen to see the error you got when running the `install_agent` task, from Litmus. If you still have it, or could re-run again to generate it, it would be great if you could share it with us. The IAC Team manages the Litmus project, so we're always keen to address any issues and improve it smiley

i created a new issue for that, see #419

**Comment by scoopex** [comment]
Feedback or merge would be nice :-)

**Comment by daianamezdrea** [comment]
Hi @scoopex, I'm currently looking at this PR! I'll test your changes and I'll come back as fast as I can with a feedback or a merge :D Sorry for the time you waited, cheers!

**Comment by scoopex** [comment]
> Hi @scoopex, I'm currently looking at this PR! I'll test your changes and I'll come back as fast as I can with a feedback or a merge :D Sorry for the time you waited, cheers!

Thanks, the changes are working now for 4 weeks on my kubernetes clusters.
Fast feedback for PR will help to improve this module dramatically ;-)

**Comment by scoopex** [comment]
Which is worse for this puppet module? No feedback for pullrequests and mergerequests to leave open forever or to apply corrections that have not been reviewed extensively? The current situation is not really satisfying for contributors.

**Comment by daianamezdrea** [comment]
Hi @scoopex, we were a little bit busy this period, but I can assure you that we value your work and the PRs from our community. I already tested 416 & 418 and the tests are green. Sorry for the long time and thank you for your patience and your contributions!

**Pull request merged**